google/cloud/storage/internal/grpc/metrics_exporter_impl.cc

```diff
@@ -63,9 +98,86 @@ absl::optional<ExporterConfig> MakeMeterProviderConfig(
                         options.get<AuthorityOption>()};
 }
 
+void EnableGrpcMetricsImpl(ExporterConfig config) {  // NOLINT
+#if !GOOGLE_CLOUD_CPP_STORAGE_AUTO_OTEL_METRICS
+  (void)config;
+#else
+  if (!ExporterRegistry::Singleton().Register(config.authority)) return;
+
+  // TODO(#13998) - the service is not ready to receive the monitored resource
+  //   and metrics defined for Google Cloud Storage clients. Erase this code
+  //   when it is.
+  auto monitored_resource =
+      config.exporter_options.get<otel_internal::MonitoredResourceOption>();
+  monitored_resource.set_type("generic_task");
+  auto& labels = *monitored_resource.mutable_labels();
+  // project_id untouched
+  // location untouched
+  labels["namespace"] = "storage_client";
+  labels["job"] = labels["host_id"];
+  labels["task_id"] = labels["instance_id"];
+  labels.erase("cloud_platform");
+  labels.erase("host_id");
+  labels.erase("instance_id");
+  labels.erase("api");
+  config.exporter_options.set<otel_internal::ServiceTimeSeriesOption>(false)
+      .set<otel_internal::MetricPrefixOption>("workload.googleapis.com/")
+      .set<otel_internal::MonitoredResourceOption>(
+          std::move(monitored_resource));
+  // END TODO(#13998) - end of code to erase.
+
+  auto exporter = otel_internal::MakeMonitoringExporter(
+      std::move(config.project),
+      monitoring_v3::MakeMetricServiceConnection(
+          std::move(config.exporter_connection_options)),
+      config.exporter_options);
+
+  auto provider = MakeGrpcMeterProvider(std::move(exporter),
+                                        std::move(config.reader_options));
+
+  auto const metrics = std::vector<absl::string_view>{
```

Comment:
err, this list will want to grow with gRPC. Is there a way to say "gimme all the metrics"? (assuming we want that)

Reply:
There is no way to say "all metrics" that I could find.  Some of these are not even listed in any header files, you just have to know they exist.